official/transformer/transformer_estimator_benchmark.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Executes Estimator benchmarks and accuracy tests."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import time
+
+from absl import flags
+from absl.testing import flagsaver
+import tensorflow as tf  # pylint: disable=g-bad-import-order
+
+from official.transformer import transformer_main as transformer_main
+from official.utils.logs import hooks
+
+TRANSFORMER_EN2DE_DATA_DIR_NAME = 'wmt32k-en2de-official'
+EN2DE_2014_BLEU_DATA_DIR_NAME = 'newstest2014'
+FLAGS = flags.FLAGS
+
+
+class EstimatorBenchmark(tf.test.Benchmark):
+  """Base class to hold methods common to test classes in the module.
+
+     Code under test for the Transformer Estimator models that report mostly the
+     same data and require the same FLAG setup.
+  """
+  local_flags = None
+
+  def __init__(self, output_dir=None, default_flags=None, flag_methods=None):
```

Comment:
You can put output_dir='/tmp' directly in the function signature, instead of having it default to None.

Note you should not do this for default_flags or flags_methods, since those default to dicts, and dicts are mutable. It's only safe to do this for immutable default values, like strings.

Reply:
I think the answer is no, because the methods that extend this end up passing None as the value and I want to catch that.  I wanted to do output_dir='/tmp' but it was not as robust.  Open to other ideas and view points.

```python
   super(TransformerBaseEstimatorAccuracy, self).__init__(
        output_dir=output_dir, flag_methods=flag_methods)
```